src/lz-db/src/transaction/tag.rs

```diff
@@ -174,6 +177,29 @@ impl Transaction<ReadWrite> {
     }
 }
 
+/// "Slugify" our tags, turning them into 7-bit alphanumeric ASCII (as well
+/// as the colon and dash).
+/// ```
+/// use lz_db;
+/// assert_eq!(lz_db::normalize_tag(&"Gödel's Incompleteness Theorem"), "godels-incompleteness-theorem");
+/// assert_eq!(lz_db::normalize_tag(&"Music::C86"), "music:c86");
+/// assert_eq!(lz_db::normalize_tag(&"  Pogo  A  Go Go!"), "pogo-a-go-go");
+/// ```
+pub fn normalize_tag<T: AsRef<str>>(tag: T) -> String {
+    // TODO: We should return Option<String> instead, allowing us to block degenerate
+    // tags such as "foo:-:bar".
+    static HYPHENIZE_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"[^a-z0-9:-]").unwrap());
+    static DEDUPE_HYPHEN_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"--+").unwrap());
+    static DEDUPE_COLON_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"::+").unwrap());
+    let normal_tag = deunicode(tag.as_ref())
+        .to_lowercase()
+        .replace(['\'', '\"'], "");
+    let normal_tag = HYPHENIZE_RE.replace_all(&normal_tag, "-");
+    let normal_tag = DEDUPE_HYPHEN_RE.replace_all(&normal_tag, "-");
+    let normal_tag = DEDUPE_COLON_RE.replace_all(&normal_tag, ":");
+    normal_tag.trim_matches('-').trim_matches(':').to_string()
+}
```

Comment:
Hrm, I'm not sure about this. Can we just add a slug field to the tags table instead? I'm currently kinda in a "let ascii die" mood, but I get that nice-looking URLs are preferable to non-nice ones...